src/units.cpp

```diff
@@ -453,18 +456,17 @@ double Units::scalingFactor(const UnitsPtr &units1, const UnitsPtr &units2, bool
     bool updateUnits2 = false;
 
     if ((units1 != nullptr) && (units2 != nullptr)) {
-        double multiplier = 0.0;
+        if ((owningModel(units1) == owningModel(units2)) && (units1->name() == units2->name())) {
+            return 1.0;
+        }
```

Comment:
I don't understand this addition, why should I care about who owns the units.  What does it have to do with the validator only checking units by name.

Reply:
It's perfectly valid to have two units in separate models with the same name.  There was a check lower down the function that returned a scaling factor of 1 if the two units had the same name.  That's only "safe" (ish) when they're from the same model.  It's not directly related to the validation stuff, but when I changed that another test broke, so this is making sure that all our current tests continue to pass even with the change.

Reply:
It was further down for a reason right.  After all else is said and done if the units have the same name then they are considered to have a scaling factor of `1.0` given this is the case by definition.

Reply:
I'm not understanding two parts of that:
> It was further down for a reason right.

There is no change to the functionality at all - the only thing that's different is that we can skip a whole lot of recursive calculation if we are going to return a value based on the name in any case.  I don't understand how doing those extra calculations is a good reason to have that test further down the function?

> After all else is said and done if the units have the same name then they are considered to have a scaling factor of 1.0 given this is the case by definition.

I don't agree.  Why would they be considered the same? We're checking units definitions here?

Reply:
I think this is still an unresolved issue and this change breaks the way this method should work